draft-ietf-masque-h3-datagram.md

```diff
@@ -169,8 +169,12 @@ Intermediaries parse the Quarter Stream ID field in order to associate the QUIC
 DATAGRAM frame with a stream. If an intermediary receives a QUIC DATAGRAM frame
 whose payload is too short to allow parsing the Quarter Stream ID field, the
 intermediary MUST treat it as an HTTP/3 connection error of type
-H3_GENERAL_PROTOCOL_ERROR. Intermediaries MUST ignore any HTTP/3 Datagram
-fields after the Quarter Stream ID.
+H3_GENERAL_PROTOCOL_ERROR. Since the Context ID is optional and its use is
+negotiated end-to-end (and extensions MAY change this negotiation in a way
+which is opaque to intermediaries), intermediaries have no way of definitely
+knowing whether the Context ID field is present or absent. Because of this,
+intermediaries MUST ignore any HTTP/3 Datagram fields after the Quarter Stream
+ID.
```

Comment:
```suggestion
H3_GENERAL_PROTOCOL_ERROR. The Context ID field is optional and its use is
negotiated end-to-end see {{ref}}. Therefore intermediaries cannot
know whether the Context ID field is present or absent and they 
MUST ignore any HTTP/3 Datagram fields after the Quarter Stream ID.
```

Here's an alternative that's less wordy at risk of losing some points you were trying to make. 

I don't think putting/repeating the negotiation requirement right here is too great, so I've replace the parenthesised clause with a reference to wherever that requirement actually lives.

Reply:
Took that text and tweaked it a bit more.